### PR TITLE
feat: Add keybind for waypoint creation

### PR DESCRIPTION
**Pull request by mbnkwm**
https://user-images.githubusercontent.com/71361356/221200969-8caf932c-079b-45b9-886d-7aa8b2a69cd5.mp4

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/map/MapFeature.java [hunk 1]
```diff
@@ -131,6 +136,15 @@ private void openMainMap() {
         McUtils.mc().setScreen(MainMapScreen.create());
     }
 
+    private void openWaypointSetup() {
+        var location = new PoiLocation(
+                McUtils.player().getBlockX(),
+                McUtils.player().getBlockY(),
+                McUtils.player().getBlockZ());
+
+        McUtils.mc().setScreen(PoiCreationScreen.create(null, location));
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/map/MapFeature.java:145]: I would like you to add a null-check in `PoiCreationScreen#onClose` to handle passing `null` as the old screen value.

**Review by kristofbolyai — COMMENTED** [review]
Otherwise LGTM, thank you :)

File: common/src/main/java/com/wynntils/features/user/map/MapFeature.java (hunk 1 above)
Comment (mbnkwm) [reply common/src/main/java/com/wynntils/features/user/map/MapFeature.java:145]: A null check there would make it impossible to leave the waypoint screen, it has to pass through to the setScreen call

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/map/MapFeature.java:145]: Right. I completely forgot that setScreen(null) means to close the screen.. This is why you don't do code reviews after a whole day of competitions..

**Review by kristofbolyai — DISMISSED** [review]

File: common/src/main/java/com/wynntils/features/user/map/MapFeature.java [hunk 2]
```diff
@@ -131,6 +136,15 @@ private void openMainMap() {
         McUtils.mc().setScreen(MainMapScreen.create());
     }
 
+    private void openWaypointSetup() {
+        var location = new PoiLocation(
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/map/MapFeature.java]:
```suggestion
        PoiLocation location = new PoiLocation(
```

We do not use var anywhere in Artemis. And we won't.

**Review by kristofbolyai — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Comment by magicus** [comment]
@mbnkwm Thank you for your contribution to Wynntils!

**Pull request merged**